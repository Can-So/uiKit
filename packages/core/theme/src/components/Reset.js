// @flow

import React, { Component, type Node } from 'react';
import styled, { css } from 'styled-components';
import * as colors from '../colors';
import { Consumer } from './Context';

type ThemeReset = {
  backgroundColor: string,
  textColor: string,
  linkColor?: string,
  linkColorHover?: string,
  linkColorActive?: string,
  linkColorOutline?: string,
  headingColor?: string,
  subtleHeadingColor?: string,
  subtleTextColor?: string,
};

const orTextColor = (preferred: string) => (p: ThemeReset) =>
  p[preferred] || p.textColor;
const Div = styled.div`
  ${p => css`
    background-color: ${p.backgroundColor};
    color: ${p.textColor};

    a {
      color: ${orTextColor('linkColor')};
    }
    a:hover {
      color: ${orTextColor('linkColorHover')};
    }
    a:active {
      color: ${orTextColor('linkColorActive')};
    }
    a:focus {
      outline-color: ${orTextColor('linkColorOutline')};
    }
    h1,
    h2,
    h3,
    h4,
    h5 {
      color: ${orTextColor('headingColor')};
    }
    h6 {
      color: ${orTextColor('subtleHeadingColor')};
    }
    small {
      color: ${orTextColor('subtleTextColor')};
    }
  `};
`;

const defaultTheme: ThemeReset = {
  backgroundColor: colors.N0,
  linkColor: colors.B400,
  linkColorHover: colors.B300,
  linkColorActive: colors.B500,
  linkColorOutline: colors.B100,
  headingColor: colors.N800,
  subtleHeadingColor: colors.N200,
  subtleTextColor: colors.N200,
  textColor: colors.N900,
};

type Props = {
  children?: Node,
};

export default class Reset extends Component<Props> {
  render() {
    return (
      <Consumer>
        {theme => (
          <Div {...defaultTheme} {...theme} {...this.props}>
            {this.props.children}
          </Div>
        )}
      </Consumer>
    );
  }
}
