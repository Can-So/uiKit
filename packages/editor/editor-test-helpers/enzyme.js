import * as tslib_1 from "tslib";
/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */
import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
// Create the IntlProvider to retrieve context for wrapping around.
var intlProvider = new IntlProvider({ locale: 'en' });
var intl = intlProvider.getChildContext().intl;
/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
    return React.cloneElement(node, { intl: intl });
}
export function shallowWithIntl(node, _a) {
    if (_a === void 0) { _a = {}; }
    var _b = _a.context, context = _b === void 0 ? {} : _b, additionalOptions = tslib_1.__rest(_a, ["context"]);
    if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
        var unwrappedType = node.type.WrappedComponent;
        node = React.createElement(unwrappedType, node.props);
    }
    return shallow(nodeWithIntlProp(node), tslib_1.__assign({ context: tslib_1.__assign({}, context, { intl: intl }) }, additionalOptions));
}
export function mountWithIntl(node, _a) {
    if (_a === void 0) { _a = {}; }
    var _b = _a.context, context = _b === void 0 ? {} : _b, _c = _a.childContextTypes, childContextTypes = _c === void 0 ? {} : _c, additionalOptions = tslib_1.__rest(_a, ["context", "childContextTypes"]);
    if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
        var unwrappedType = node.type.WrappedComponent;
        node = React.createElement(unwrappedType, node.props);
    }
    return mount(nodeWithIntlProp(node), tslib_1.__assign({ context: tslib_1.__assign({}, context, { intl: intl }), childContextTypes: tslib_1.__assign({}, childContextTypes, { intl: intlShape }) }, additionalOptions));
}
//# sourceMappingURL=enzyme.js.map