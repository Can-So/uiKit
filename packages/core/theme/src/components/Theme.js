// @flow

import React, { Component, type Node } from 'react';
import { Consumer, Provider } from '../components/Context';

type Props = {
  /**
   * Children to pass to the theme. If this is a function, the `theme`
   * function is called and passed to the children function. If it is anything
   * else, the `theme` function is called and set as context for all
   * descendants.
   */
  children: ((*) => Node) | Node,

  /**
   * The theme function that is called with the parent theme and returns the
   * theme values.
   */
  theme: (*) => *,

  /**
   * DEPRECATED. Alias for `theme`.
   */
  values?: (*) => *,
};

export default class Theme extends Component<Props> {
  static defaultProps = {
    theme: (v: *): * => v,
  };
  render() {
    const { children, theme, values } = this.props;
    return (
      <Consumer>
        {parent => {
          const merged = (values || theme)(parent);
          return typeof children === 'function' ? (
            children(merged)
          ) : (
            <Provider value={merged}>{children}</Provider>
          );
        }}
      </Consumer>
    );
  }
}
