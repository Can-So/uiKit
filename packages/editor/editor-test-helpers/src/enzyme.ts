/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */
import * as React from 'react';
import { ReactElement } from 'react';
import { IntlProvider, InjectedIntlProps, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' });
const { intl } = intlProvider.getChildContext();

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl<P>(
  node: ReactElement<P>,
  { context = {}, ...additionalOptions } = {},
) {
  if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
    const unwrappedType = (node.type as any).WrappedComponent;
    (node as any) = React.createElement(unwrappedType, node.props);
  }
  return shallow(
    nodeWithIntlProp(node) as ReactElement<P & InjectedIntlProps>,
    {
      context: { ...context, intl },
      ...additionalOptions,
    },
  );
}

export function mountWithIntl<P>(
  node: ReactElement<P>,
  { context = {}, childContextTypes = {}, ...additionalOptions } = {},
) {
  if (typeof node.type !== 'string' && node.type.name === 'InjectIntl') {
    const unwrappedType = (node.type as any).WrappedComponent;
    (node as any) = React.createElement(unwrappedType, node.props);
  }
  return mount(nodeWithIntlProp(node) as ReactElement<P & InjectedIntlProps>, {
    context: { ...context, intl },
    childContextTypes: { ...childContextTypes, intl: intlShape },
    ...additionalOptions,
  });
}
