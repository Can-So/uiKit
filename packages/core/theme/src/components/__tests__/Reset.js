// @flow

import React from 'react';
import { mount } from 'enzyme';
import Reset from '../Reset';
import Theme from '../Theme';

test('reset', () => {
  expect(mount(<Reset />)).toMatchSnapshot();
});

test('themed reset', () => {
  expect(
    mount(
      <Theme
        theme={() => ({
          backgroundColor: '#000',
          textColor: '#fff',
          linkColor: 'blue',
        })}
      >
        <Reset />
      </Theme>,
    ),
  ).toMatchSnapshot();
});
