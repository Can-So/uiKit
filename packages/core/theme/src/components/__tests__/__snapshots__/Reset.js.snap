// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset 1`] = `
.c0 {
  background-color: #FFFFFF;
  color: #091E42;
}

.c0 a {
  color: #0052CC;
}

.c0 a:hover {
  color: #0065FF;
}

.c0 a:active {
  color: #0747A6;
}

.c0 a:focus {
  outline-color: #4C9AFF;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5 {
  color: #172B4D;
}

.c0 h6 {
  color: #6B778C;
}

.c0 small {
  color: #6B778C;
}

<Reset>
  <styled.div
    backgroundColor="#FFFFFF"
    headingColor="#172B4D"
    linkColor="#0052CC"
    linkColorActive="#0747A6"
    linkColorHover="#0065FF"
    linkColorOutline="#4C9AFF"
    subtleHeadingColor="#6B778C"
    subtleTextColor="#6B778C"
    textColor="#091E42"
  >
    <div
      className="c0"
    />
  </styled.div>
</Reset>
`;

exports[`themed reset 1`] = `
.c0 {
  background-color: #FFFFFF;
  color: #091E42;
}

.c0 a {
  color: #0052CC;
}

.c0 a:hover {
  color: #0065FF;
}

.c0 a:active {
  color: #0747A6;
}

.c0 a:focus {
  outline-color: #4C9AFF;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5 {
  color: #172B4D;
}

.c0 h6 {
  color: #6B778C;
}

.c0 small {
  color: #6B778C;
}

<Theme
  backgroundColor="#000"
  linkColor="blue"
  textColor="#fff"
  values={Object {}}
>
  <Reset>
    <styled.div
      backgroundColor="#FFFFFF"
      headingColor="#172B4D"
      linkColor="#0052CC"
      linkColorActive="#0747A6"
      linkColorHover="#0065FF"
      linkColorOutline="#4C9AFF"
      subtleHeadingColor="#6B778C"
      subtleTextColor="#6B778C"
      textColor="#091E42"
    >
      <div
        className="c0"
      />
    </styled.div>
  </Reset>
</Theme>
`;
